src/tyr/serializers.cc

```diff
@@ -34,17 +34,23 @@ midgard::PointLL to_ll(const LatLng& ll) {
 } // namespace
 namespace valhalla {
 namespace tyr {
-json::ArrayPtr route_references(const TripRoute& route, const Options& options) {
+void route_references(json::MapPtr& route_json, const TripRoute& route, const Options& options) {
+  const bool linear_reference =
+      options.linear_references() &&
+      (options.action() == Options::trace_route || options.action() == Options::route);
+  if (!linear_reference) {
+    return;
+  }
   if (options.costing() != Costing::auto_) {
```

Comment:
We should remove this if, I know that frc will potentially be mostly 7 but that's ok